src/frontend/sharedTypes.ts

```diff
@@ -39,3 +41,9 @@ export type NavigationProp = NavigationStackProp<
 export type StackScreenComponent = NavigationStackScreenComponent<
   NavigationParams
 >;
+
+type NavKeys = keyof AppStackList;
+
+export type NativeNavigationProp<
+  ScreenName extends NavKeys
+> = NativeStackScreenProps<AppStackList, ScreenName>;
```

Comment:
This is used to define the props of navigation screens (that are received from the navigator).